<a href="https://colab.research.google.com/github/alina-bogdanovskaya/Probability_Statistics_Dec-2021/blob/lesson8/ANOVA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
%matplotlib inline
import numpy as np
from scipy import stats

Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

H0: μ1 = μ2 = μ3

H1: μ1 $\neq$ μ2 $\neq$ μ3

In [2]:
f = np.array([173, 175, 180, 178, 177, 185, 183, 182])
h = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
w = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

In [3]:
stats.f_oneway(f, h, w)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)

α = 0.05 

0.05 > 0.0105, следовательно верна гипотеза H1, и различие между группами статистически значимо 

α = 0.01

0.01 < 0.0105, следовательно верна гипотеза H0, и не существует статистически значимого различия между группами

In [4]:
k = 3

n1 = f.size
n2 = h.size
n3 = w.size

n = n1 + n2 + n3

In [5]:
f_mean = np.mean(f)
f_mean

179.125

In [6]:
h_mean = np.mean(h)
h_mean

178.66666666666666

In [7]:
w_mean = np.mean(w)
w_mean

172.72727272727272

In [8]:
total = np.concatenate([f, h, w])
total_mean = np.mean(total)
total_mean

176.46428571428572

In [9]:
S_F = ((f_mean - total_mean)**2) * n1 + ((h_mean - total_mean)**2) * n2 + ((w_mean - total_mean)**2) * n3
S_F

253.9074675324678

In [10]:
S_r = ((f - f_mean)**2).sum() + ((h - h_mean)**2).sum() + ((w - w_mean)**2).sum()
S_r

577.0568181818182

In [11]:
S = ((total - total_mean)**2).sum()
S

830.9642857142854

In [13]:
S_F + S_r

830.964285714286

In [14]:
var_F = S_F/(k-1)
var_F

126.9537337662339

In [15]:
var_r = S_r/(n-k)
var_r

23.08227272727273

In [16]:
F_с = var_F/var_r
F_с

5.500053450812598

α = 0.05 => F = 3.39

5.5 > 3.39, следовательно различие между группами статистически значимо

α = 0.01 => F = 5.57

5.5 < 5.57, следовательно верна гипотеза Н0, и статистически значимого различия между группами не существует 
